src/lib/decode-conversions.ts

```diff
@@ -187,6 +194,55 @@ export const convertCoreOwnership: ConvertFunction<'coreOwnership'> = (
   }
 }
 
+export const convertIcon: ConvertFunction<'icon'> = (message, versionObj) => {
+  const { common, schemaVersion, ...rest } = message
+  const jsonSchemaCommon = convertCommon(common, versionObj)
+  return {
+    ...jsonSchemaCommon,
+    ...rest,
+    variants: message.variants.map((variant) => convertIconVariant(variant)),
+  }
+}
+
+function convertIconVariant(variant: Icon_1_IconVariant) {
+  const { blobVersionId, mimeType, size, pixelDensity } = variant
+  return {
+    blobVersionId: blobVersionId.toString('hex'),
+    mimeType: convertIconMimeType(mimeType) as ValidMimeTypes,
```

Comment:
```suggestion
    mimeType: convertIconMimeType(mimeType),
```
as far as i can tell this is unnecessary